Northstar.CustomServers/mod/scripts/vscripts/mp/_gamestate_mp.nut

```diff
@@ -868,3 +878,126 @@ float function GetTimeLimit_ForGameMode()
 	// default to 10 mins, because that seems reasonable
 	return GetCurrentPlaylistVarFloat( playlistString, 10 )
 }
+
+// faction dialogue
+void function KilledPlayerTitanDialogue( entity attacker, entity victim )
+{
+	if( !attacker.IsPlayer() )
+		return
+	entity titan
+	if ( victim.IsTitan() )
+		titan = victim
+
+	if( !IsValid( titan ) )
+		return
+	string titanCharacterName = GetTitanCharacterName( titan )
+
+	switch( titanCharacterName )
+	{
+		case "ion":
+			PlayFactionDialogueToPlayer( "kc_pilotkillIon", attacker )
+			return
+		case "tone":
+			PlayFactionDialogueToPlayer( "kc_pilotkillTone", attacker )
+			return
+		case "legion":
+			PlayFactionDialogueToPlayer( "kc_pilotkillLegion", attacker )
+			return
+		case "scorch":
+			PlayFactionDialogueToPlayer( "kc_pilotkillScorch", attacker )
+			return
+		case "ronin":
+			PlayFactionDialogueToPlayer( "kc_pilotkillRonin", attacker )
+			return
+		case "northstar":
+			PlayFactionDialogueToPlayer( "kc_pilotkillNorthstar", attacker )
+			return
+		default:
+			PlayFactionDialogueToPlayer( "kc_pilotkilltitan", attacker )
+			return
+	}
+}
+
+void function DialoguePlayNormal()
+{
+	int totalScore = GameMode_GetScoreLimit( GameRules_GetGameMode() )
+	int winningTeam
+	int losingTeam
+	float diagIntervel = 70 // play a faction dailogue every 70s
+
+	while( GetGameState() == eGameState.Playing )
+	{
+		wait diagIntervel
+		if( GameRules_GetTeamScore( TEAM_MILITIA ) < GameRules_GetTeamScore( TEAM_IMC ) )
+		{
+			winningTeam = TEAM_IMC
+			losingTeam = TEAM_MILITIA
+		}
+		if( GameRules_GetTeamScore( TEAM_MILITIA ) > GameRules_GetTeamScore( TEAM_IMC ) )
+		{
+			winningTeam = TEAM_MILITIA
+			losingTeam = TEAM_IMC
+		}
+		if( GameRules_GetTeamScore( winningTeam ) - GameRules_GetTeamScore( losingTeam ) >= totalScore * 0.4 )
+		{
+			PlayFactionDialogueToTeam( "scoring_winningLarge", winningTeam )
+			PlayFactionDialogueToTeam( "scoring_losingLarge", losingTeam )
+		}
+		else if( GameRules_GetTeamScore( winningTeam ) - GameRules_GetTeamScore( losingTeam ) <= totalScore * 0.2 )
+		{
+			PlayFactionDialogueToTeam( "scoring_winningClose", winningTeam )
+			PlayFactionDialogueToTeam( "scoring_losingClose", losingTeam )
+		}
+		else if( GameRules_GetTeamScore( winningTeam ) == GameRules_GetTeamScore( losingTeam ) )
+		{
+			continue
+		}
+		else
+		{
+			PlayFactionDialogueToTeam( "scoring_winning", winningTeam )
+			PlayFactionDialogueToTeam( "scoring_losing", losingTeam )
+		}
+	}
+}
+
+void function DialoguePlayWinnerDetermined()
+{
+	if( GAMETYPE != "ctf" && IsRoundBased() || IsFFAGame() ) // such modes shouldn't have winner dialogue
```

Comment:
shouldn't hardcode ctf here, this should be modular

Reply:
tried to fix this